cmd/main.go

```diff
@@ -299,11 +304,101 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Initialize metrics
+	logger.Log.Infof("Creating metrics emitter instance")
+	// Force initialization of metrics by creating a metrics emitter
+	_ = metrics.NewMetricsEmitter()
+	logger.Log.Infof("Metrics emitter created successfully")
+
+	// Configure Prometheus client using flexible configuration with TLS support
+	promConfig, err := utils.GetPrometheusConfig(context.Background(), mgr.GetClient())
+	if err != nil {
+		setupLog.Error("failed to get Prometheus configuration", zap.Error(err))
+		os.Exit(1)
+	}
+
+	// ensure we have a valid configuration
+	if promConfig == nil {
+		setupLog.Error("no Prometheus configuration found - this should not happen")
+		os.Exit(1)
+	}
+
+	// Always validate TLS configuration since HTTPS is required
+	if err := utils.ValidateTLSConfig(promConfig); err != nil {
+		logger.Log.Errorf("TLS configuration validation failed - HTTPS is required: error=%v", err)
+		setupLog.Error("TLS configuration validation failed", zap.Error(err))
+		os.Exit(1)
+	}
+
+	logger.Log.Infof("Initializing Prometheus client -> address: %s, tls_enabled: true", promConfig.BaseURL)
+
+	// Create Prometheus client with TLS support
+	promClientConfig, err := utils.CreatePrometheusClientConfig(promConfig)
+	if err != nil {
+		setupLog.Error("failed to create prometheus client config", zap.Error(err))
+		os.Exit(1)
+	}
+
+	promClient, err := api.NewClient(*promClientConfig)
+	if err != nil {
+		setupLog.Error("failed to create prometheus client", zap.Error(err))
+		os.Exit(1)
+	}
+
+	promAPI := promv1.NewAPI(promClient)
+
+	// Validate that the API is working by testing a simple query with retry logic
+	if err := utils.ValidatePrometheusAPI(context.Background(), promAPI); err != nil {
+		logger.Log.Errorf("CRITICAL: Failed to connect to Prometheus - Inferno requires Prometheus connectivity for autoscaling decisions: error=%v", err)
```

Comment:
what's Inferno?

Reply:
Ok, that's what we called WVA in the early days, inferno-autoscaler. The greedy optimizer is still called Inferno. A few log files need to be changed. Thanks, I will change it.